app/controllers/questions_controller.rb

```diff
@@ -29,11 +29,26 @@ def show
     @answer = @question.answers.new
   end
 
+  def edit
+    @question = Question.find(params[:id])
+    @categories = Category.all
+  end
+
+  def update
+    @question = Question.find(params[:id])
+    if @question.update(question_params)
+      flash[:success] = "Your question was edited successfully!"
+      redirect_to question_path(@question)
+    else
+      flash[:danger] = "Failed to edit question. Please try again."
```

Comment:
It might be possible to add a conditional that looks at question_params to see what is missing and add individual error messages depending on what is missing.